### ISSUE TITLE
(Feature) Raise DynDNS rate limit to 2 requests per minute or per 2 minutes

### ISSUE THREAD

**Issue by eomanis** [desec-io/desec-stack#806]
Currently, when a user updates their IPv4 and IPv6 address separately yet chronologically immediately adjacent, the second update request fails due to [having a rate limit of 1 request per minute and domain](https://desec.readthedocs.io/en/latest/rate-limits.html#api-request-throttling).

While in the overwhelming majority of cases the IPv4 and IPv6 addresses change at the same time (think WAN reconnect) and then can both be updated with a single request, this may not always be the case in every situation.

Also, even when they change at the same time, there may be legitimate reasons why a user may run separate DynDNS clients for IPv4 and IPv6, possibly on different machines, which may often cause one of the clients to suffer a rate limit block.
See [this forum thread](https://talk.desec.io/t/raise-dyndns-rate-limit-to-2-requests-per-minute-or-per-2-minutes/684/2) for more in-depth reasoning.

Basically, IPv4 and IPv6 being entirely separate protocols, I argue that updating the IPv4 and IPv6 records in one request is an optimization for an – admittedly very common – special case, and doing separate updates is the general/universal case, and the general case should be accommodated for as well.

**Comment by girlbossceo** [comment]
Just want to echo what was said above with my problem / use-case with this. 1 request per minute is very cumbersome when you have even just a few domains that require you to update each A record every 60 seconds. This is what I currently have to do to workaround this very aggressive API limit (I don't want to use a dynamic DNS client for minimalism):

```
ExecStart=/usr/bin/curl --tlsv1.3 --user kennel.girlcock.ceo:<redacted> 'https://update.dedyn.io/'
ExecStart=/bin/sleep 60
ExecStart=/usr/bin/curl --tlsv1.3 --user qbt.kennel.girlcock.ceo:<redacted> 'https://update.dedyn.io/'
ExecStart=/bin/sleep 60
ExecStart=/usr/bin/curl --tlsv1.3 --user cockpit.kennel.girlcock.ceo:<redacted> 'https://update.dedyn.io/'
ExecStart=/bin/sleep 60
ExecStart=/usr/bin/curl --tlsv1.3 --user synadm.kennel.girlcock.ceo:<redacted> 'https://update.dedyn.io/'
ExecStart=/bin/sleep 60
ExecStart=/usr/bin/curl --tlsv1.3 --user ytdl.kennel.girlcock.ceo:<redacted> 'https://update.dedyn.io/'
ExecStart=/bin/sleep 60
ExecStart=/usr/bin/curl --tlsv1.3 --user nextcloud.kennel.girlcock.ceo:<redacted> 'https://update.dedyn.io/'
ExecStart=/bin/sleep 60
ExecStart=/usr/bin/curl --tlsv1.3 --user vault.kennel.girlcock.ceo:<redacted> 'https://update.dedyn.io/'
```

Something like this feels very hacky for when I'm simply changing the IP for all of those domains to the exact same IP and protocol and record type, just with 60 second intervals in between each request.

**Comment by peterthomassen** [comment]
@girlbossceo, your use case smells like a perfect for for CNAME records. Set `*.kennel.girlcock.com CNAME kennel.girlcock.com`, and you will only need to do the first update request. All subdomains will benefit from this single update.

This solution is cleaner, more efficient, less error-prone and wasteful, ... than updating all subdomains separately. It's not clear why we should devote resources in enabling the less clean, less efficient etc. solution.

For the record, I do have sympathy for some of the arguments in https://github.com/desec-io/desec-stack/issues/806#issue-1897108216, and we will accept pull requests that implement per-domain+protocol rate limiting.

**Comment by girlbossceo** [comment]
@peterthomassen I'm personally not a fan of wildcard records, especially since my CAA record forbids wildcard issuance. Wildcard certificates require more care and security to prevent abuse in the event of a compromised server.

I could maybe do that just with individual subdomains pointing to the CNAME, but this still feels like it shouldn't be aggressively ratelimiting for the same IP and record type.

Nonetheless, I am at least glad to hear that this kind of feature is open for implementation, even if some use-cases could be optimised.

**Comment by peterthomassen** [comment]
> I could maybe do that just with individual subdomains pointing to the CNAME

Yes, that's a good idea. The wildcard was mainly for illustration of the CNAME concept. Of course, explicit is better.

**Comment by JeGr** [comment]
I wanted to chime in on that topic as it seems I've run into this, too, and it's not a thing, that can be mitigated by CNAMEs for me and as the rate limit is per domain, I can't avoid it.

I'm updating a few records from my own domain every morning, as my _insert-swear-word-here_ DSL providers still insists on daily disconnects. Thus I'm hitting the 1 dyndns call per domain per minute limit. As the updating entity is my firewall and it can't update v4 and v6 at the same time, those are two entries (with the preserve parameter) and with IPv6 coming first, I've encountered it multiple times, that the v4 entry was skipped due to 429 errors afterwards. So now I have to test and reconnect to the DynDNS API again and again after disconnect every day for multiple times so all dyndns entries get set up correctly (one v4 only, one v6 only, one with both). If the API allowed for 3-5 hits per domain, that wouldn't be neccessary. 
And as I can't update those entries in another way I was wondering if upping the limit is completely out of question?

Cheers
\jens
